### ISSUE TITLE
Check `case_panels` stored on variant collection

### ISSUE THREAD

**Issue by dnil** [Clinical-Genomics/scout#3857]
`case_panels` were as far as I remember designed to be used in display populated versions of the variants, not to be stored back to db. This is not a huge issue, but variants are plentiful, and I don't quite see the need to save this permanently, when it anyway has to be rechecked each time the variant is to be displayed. I rather suspect I either made a mistake in that PR (#3837?) and/or that we have an older issue with variants getting re-written to db on load (perhaps like an overzealous re-check of compounds writing the whole in memory variant, rather than a select portion of it?).

**Comment by dnil** [comment]
The best current theory is that we have some overzealous writing of rather fully populated `variant_obj` to db from here  https://github.com/Clinical-Genomics/scout/blob/e9ce44b0f61b504b79c75f212f3c61d9139a6917/scout/server/blueprints/variants/controllers.py#L781

**Comment by northwestwitch** [comment]
Remove also all the str variant - related keys with null values whenever cleaning the docs (see #4288)

**Comment by northwestwitch** [comment]
I think we might have fixed this issue somehow already? Checked [strongbison](https://scout-stage.scilifelab.se/cust000/15026-mip7), which has been re-uploaded on Monday to test the new release. When I go to variantS page I don't see `case_panels` in one of these variants any more. So I'm going to close this issue, but let's keep an eye on the variant fields in the future.

**Comment by dnil** [comment]
I also could not reproduce it locally; I guess the question remains if we should scan for variants with it?

**Comment by northwestwitch** [comment]
There are a bunch in [this case](https://scout.scilifelab.se/cust010/A1319) among the first results if you search prod variants with the query: `{case_panels:{"$exists":true}}`

**Comment by dnil** [comment]
Yes, but also more current ones, e.g. [this case](https://scout.scilifelab.se/cust002/F0052376).

**Comment by dnil** [comment]
I've tried getting a count, but so far I've only managed to hit slow queries and so timeouts.

**Comment by northwestwitch** [comment]
> I've tried getting a count, but so far I've only managed to hit slow queries and so timeouts.

same!

**Comment by dnil** [comment]
From a random sampling, I've seen variants with interactions (like dismiss) but also some that have no apparent interaction (compounds?).

**Comment by northwestwitch** [comment]
Reopening since we don't fully understand why this happens yet. Only a few variants from a case are affected. Possibly the error is here: https://github.com/Clinical-Genomics/scout/blob/e9ce44b0f61b504b79c75f212f3c61d9139a6917/scout/server/blueprints/variants/controllers.py#L781

**Comment by dnil** [comment]
Yes, at least that is one place. There are two options I can see: one is to do a selective variant update, setting the fields that may have changed, instead of upserting the whole variant_obj. The other is to enforce the variant object model for storage, e.g. with a variant class, perhaps a pydantic one. But that is a decently big rewrite.

**Comment by dnil** [comment]
Or we could do it the cleanup route, like for case obj, removing keys that have no business on the variant obj.
